### PR TITLE
EDGOAIPMH-13 Validate Query Arguments

### PR DESCRIPTION
**Pull request by epam-pavelk**
OAI-PMH request parameters validation logic

### CODE REVIEW

File: src/main/java/org/folio/edge/oaipmh/Verb.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+import static org.folio.edge.oaipmh.Constants.SET;
+import static org.folio.edge.oaipmh.Constants.UNTIL;
+import static org.folio.edge.oaipmh.Constants.VERB;
+import static org.openarchives.oai._2.OAIPMHerrorcodeType.BAD_ARGUMENT;
+
+/**
+ * Enum that represents OAI-PMH verbs with associated http parameters and validation logic.
+ */
+public enum Verb {
+  GET_RECORD("GetRecord", of(IDENTIFIER, METADATA_PREFIX), EMPTY_SET, null),
```
Comment (craigmcnally) [inline src/main/java/org/folio/edge/oaipmh/Verb.java]: I'm not familiar with Guava, so this is new to me... pretty convenient!

File: src/main/java/org/folio/edge/oaipmh/OaiPmhHandler.java [hunk 2]
```diff
@@ -1,17 +1,90 @@
...
+    }
+  }
+
+  @Override
+  protected void badRequest(RoutingContext ctx, String body) {
+    badRequest(ctx, body, BAD_ARGUMENT);
+  }
+
+  private void badRequest(RoutingContext ctx, String body, OAIPMHerrorcodeType type) {
+    OAIPMH resp = new OAIPMH().withErrors(new OAIPMHerrorType().withCode(type).withValue(body));
```
Comment (piotr-kalashuk) [inline src/main/java/org/folio/edge/oaipmh/OaiPmhHandler.java]: According to the [specification](https://www.openarchives.org/OAI/openarchivesprotocol.html#XMLResponse) the OAI-PMH response must always contain responseDate and request elements. Better to add them now as well

Comment (craigmcnally) [reply src/main/java/org/folio/edge/oaipmh/OaiPmhHandler.java]: Agreed.  Good catch @piotr-kalashuk

File: src/main/java/org/folio/edge/oaipmh/Verb.java (hunk 1 above)
Comment (epam-pavelk) [reply src/main/java/org/folio/edge/oaipmh/Verb.java]: Yeah, Java 8 still missing pretty known and convenient features. Java 9 though introduced these convenience factory methods for collections.

File: src/main/java/org/folio/edge/oaipmh/OaiPmhHandler.java (hunk 2 above)
Comment (epam-pavelk) [reply src/main/java/org/folio/edge/oaipmh/OaiPmhHandler.java]: Required elements are added.

As for baseUrl, I populate it using the following code:
String baseUrl = StringUtils.substringBefore(ctx.request().absoluteURI(), "?");

@craigmcnally what do you think?

Comment (craigmcnally) [reply src/main/java/org/folio/edge/oaipmh/OaiPmhHandler.java]: Yes, I think that makes sense

**Pull request merged**